### ISSUE TITLE
EnergyPlus crashes when assigning AirLoopHVAC:UnitarySystem as zone equipment

### ISSUE THREAD

**Issue by aaron-boranian** [NREL/EnergyPlus#5287]
The InputOutputReference.pdf for v8.4 states that "this object can be modeled as zone equipment (i.e., listed in a ZoneHVAC:EquipmentList) or as an outside air system component (i.e., listed in a AirLoopHVAC:OutdoorAirSystem:EquipmentList)." I took the 'UnitarySystem_DXCoilSystemAuto.idf' example file and added another instance of the AirLoopHVAC:UnitarySystem object to connect to a zone through it's referenced ZoneHVAC:EquipmentList object. When I tried to simulate, there was a hard EnergyPlus crash after the sizing periods once the annual simulation began. You can download the file from Google Drive [HERE](https://drive.google.com/a/bigladdersoftware.com/file/d/0B_SyIwm6jC9MazZTY2d4bG9JaVk/view?usp=sharing). 

Looking through example files that include the AirLoopHVAC:UnitarySystem object, they all place them on Branch objects for the main air handler, none of them place them in ZoneHVAC:EquipmentList objects. Also, in the v8.4 IDD file the AirLoopHVAC:UnitarySystem is not included in the list of acceptable object types for ZoneHVAC:EquipmentList.

**Comment by rraustad** [comment]
The AirloopHVAC:UnitarySystem was intended to be used as zone equipment and has been minimally tested as such. For the configuration tested, it did work. The IDD probably missed this nuance. If there is an issue with using UnitarySystem as zone equipment, then yes an issue is warranted.

**Comment by aaron-boranian** [comment]
What do you mean by "for the configuration tested, it did work"? 

If the IDD missed this capability, then I'm glad I caught this disconnect between the InputOutputReference.pdf so that it can be addressed for the next version release (hopefully).

**Comment by rraustad** [comment]
While developing the UnitarySystem model, I created a test file that had three (3) AirloopHVAC:UnitarySystem objects. One in the AirLoopHVAC:OutdoorAirSystem:EquipmentList, one in the AirLoopHVAC main branch, and one in a ZoneHVAC:EquipmentList. I just now ran it in V8.4 and it still runs successfully. That is the extent of testing UnitarySystem as zone equipment. I assume there is something different between your defect file and the one I am running (not surprisingly). The point is UnitarySystem should work as zone equipment.

**Comment by rraustad** [comment]
I just ran the defect file and would consider this an issue. The problem is due to using both an air loop UnitarySystem and zone equipment UnitarySystem in the same zone.

```
ZoneHVAC:EquipmentList,
  Zone3Equipment,          !- Name
  AirTerminal:SingleDuct:Uncontrolled,  !- Zone Equipment 1 Object Type
  Zone3DirectAir,          !- Zone Equipment 1 Name
  1,                       !- Zone Equipment 1 Cooling Sequence
  1,                       !- Zone Equipment 1 Heating or No-Load Sequence
  AirLoopHVAC:UnitarySystem,  !- Zone Equipment 1 Object Type
  NORTH Zone Heating Coil,          !- Zone Equipment 1 Name
  2,                       !- Zone Equipment 1 Cooling Sequence
  2;                       !- Zone Equipment 1 Heating or No-Load Sequence
```

**Comment by mjwitte** [comment]
@rraustad Seems that should be OK as long as it's not the same unitarysystem object referenced in both places - need protection against that if it's not already checked.

**Comment by rraustad** [comment]
Agreed. Working now, just testing out a few things.

**Comment by rraustad** [comment]
This did take a few lines of code changes and additions but it is working now. 

![5287_fixed](https://cloud.githubusercontent.com/assets/5778156/11245524/2cbb5792-8de1-11e5-9bc6-7986c2956bb1.png)

**Comment by JasonGlazer** [comment]
Closed from #5331

**Comment by rraustad** [comment]
Thank you Jason.
